lib/fizzy/parser.cpp

```diff
@@ -35,208 +32,182 @@ std::tuple<bool, const uint8_t*> parseGlobalType(const uint8_t* pos)
     return {is_mutable, pos};
 }
 
-template <>
-struct parser<ConstantExpression>
+inline parser_result<ConstantExpression> parse_constant_expression(const uint8_t* pos)
```

Comment:
This is not a template specialization as it is never used in `parse_vec()`. But we can make it so if easier to read.